codemon/CodemonInit.py

```diff
@@ -6,8 +6,26 @@ def write_to_file(filename, text, contestName=None):
   full_filename = os.path.join(os.getcwd(), filename)
   if contestName is not None:
     full_filename = os.path.join(os.getcwd(), contestName, filename)
-  with open(full_filename, 'w+') as f:
-    f.write(text)
+
+    os.makedirs(full_filename)
+    
+    cppFile = full_filename + '/' + filename + '.cpp'
+    f = open(cppFile,"w+")
+    template = template_cpp()
+    f.write(template)
+
+    inpFile = full_filename + '/' + filename + '.in'
+    f = open(inpFile,"w+")
+    outFile = full_filename + '/' + filename + '.op'
+    f = open(outFile,"w+")
+
+    cppTest = full_filename + '/' + 'test_case.cpp'
+    f = open(cppTest,"w+")
+    inpTest = full_filename + '/' + 'test_case.in'
+    f = open(inpTest,"w+")
+    outTest = full_filename + '/' + 'test_case.out'
+    f = open(outTest,"w+")
+

```

Comment:
- We don't want to repeat the same code so many times, also these operations should not be in this function. This function is a helper to `init` and that's where these operations should take place. 
- Dont use `full_filename` + '/' + 'test_case.in', instead use functions like `os.path.join()`. Please refactor here.
- Look into the code of this file properly, test it properly and then refactor it as much as possible.

Reply:
Tried to solve this issue, have a look at it and let me know how can I improve it more.
Thanks.